pkg/client/icinga/config.go

```diff
@@ -40,21 +42,24 @@ func getIcingaSecretData(kubeClient clientset.Interface, secretName string) (*au
 	}
 
 	authData := new(authInfo)
-	if data, found := secret.Data[env]; found {
+	if data, found := secret.Data[ENV]; found {
 		dataReader := strings.NewReader(string(data))
 		secretData, err := ini.Load(dataReader)
 		if err != nil {
 			return nil, err
 		}
 
+		host, found := secretData.Get("", IcingaService)
```

Comment:
You need to check if the host is a IP or not. Call `net.ParseIP(host)`, if returns not null, then you have an IP address.